### PR TITLE
fixed zabbix_action to search for mediatype by its name in Zabbix 4.4+

### LINKED ISSUE #75: zabbix_action: media_type in operations does not choose the right one
##### SUMMARY

Hi,
i have created a new mediatype named BMC_Notification.
within my playbook i have created a new action named "Send alerts to BMC".
i have chosen the existing media_type BMC_Notification in this task, but when it was running i can see always that the mediatype "Email" was set.

i asume that the lines in the module
```
            return {
                'default_msg': '0' if operation.get('message') is not None or operation.get('subject')is not None else '1',
                'mediatypeid': self._zapi_wrapper.get_mediatype_by_mediatype_name(
                    operation.get('media_type')
                ) if operation.get('media_type') is not None else '0',
                'message': operation.get('message'),
                'subject': operation.get('subject'),
            }
```
returns a wrong id for the given media_type

In the Zabbix gui i can choose the right media_type in the ansible created action. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

zabbix_action
( https://github.com/ansible-collections/community.zabbix/blob/master/plugins/modules/zabbix_action.py )

##### ANSIBLE VERSION

```
ansible 2.9.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/admin/pweibbre/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 20 2019, 13:30:36) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36.0.1)]
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

Create a media_type and run the task.

```
- name: deploy trigger action for BMC notification
  zabbix_action:
    login_user: Admin
    login_password: "{{ zabbix_admin_pw }}"
    server_url: "{{ zabbix_url }}"
    state: present
    event_source: 'trigger'
    status: enabled
    name: "Send alerts to BMC"
    esc_period: 1h
    conditions:
      - type: 'trigger_name'
        operator: 'like'
        value: 'WebCheck'
    operations:
      - type: send_message
        media_type: 'BMC_Notification'
        send_to_users:
          - 'zbx_robot'
    default_subject: 'Problem: {EVENT.NAME}'
    default_message: |
      Problem started at {EVENT.TIME} on {EVENT.DATE}
      Problem name: {EVENT.NAME}
      Host: {HOST.NAME}
      Severity: {EVENT.SEVERITY}

      Original problem ID: {EVENT.ID}
      {TRIGGER.URL}
    recovery_default_subject: 'Resolved: {EVENT.NAME}'
    recovery_default_message: |
      Problem has been resolved at {EVENT.RECOVERY.TIME} on {EVENT.RECOVERY.DATE}
      Problem name: {EVENT.NAME}
      Host: {HOST.NAME}
      Severity: {EVENT.SEVERITY}

      Original problem ID: {EVENT.ID}
      {TRIGGER.URL}
    recovery_operations:
      - type: send_message
        media_type: 'BMC_Notification'
        send_to_users:
          - 'zbx_robot'
    acknowledge_default_subject: 'Updated problem: {EVENT.NAME}'
    acknowledge_default_message: |
      {USER.FULLNAME} {EVENT.UPDATE.ACTION} problem at {EVENT.UPDATE.DATE} {EVENT.UPDATE.TIME}.
      {EVENT.UPDATE.MESSAGE}

      Current problem status is {EVENT.STATUS}, acknowledged: {EVENT.ACK.STATUS}
    validate_certs: false
```

##### EXPECTED RESULTS

Playbook should set mediatype for this new action to the configured one ( BMC_Notification )

##### ACTUAL RESULTS
Playbook runs without errors, but the wrong mediatype is set in Zabbix.
Alwys set to Email instead of the configured one.

```
ok: [hostname] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "acknowledge_default_message": "{USER.FULLNAME} {EVENT.UPDATE.ACTION} problem at {EVENT.UPDATE.DATE} {EVENT.UPDATE.TIME}.\n{EVENT.UPDATE.MESSAGE}\n\nCurrent problem status is {EVENT.STATUS}, acknowledged: {EVENT.ACK.STATUS}\n",
            "acknowledge_default_subject": "Updated problem: {EVENT.NAME}",
            "acknowledge_operations": [],
            "conditions": [
                {
                    "formulaid": null,
                    "operator": "like",
                    "type": "trigger_name",
                    "value": "WebCheck",
                    "value2": null
                }
            ],
            "default_message": "Problem started at {EVENT.TIME} on {EVENT.DATE}\nProblem name: {EVENT.NAME}\nHost: {HOST.NAME}\nSeverity: {EVENT.SEVERITY}\n\nOriginal problem ID: {EVENT.ID}\n{TRIGGER.URL}\n",
            "default_subject": "Problem: {EVENT.NAME}",
            "esc_period": "1h",
            "eval_type": null,
            "event_source": "trigger",
            "formula": null,
            "http_login_password": null,
            "http_login_user": null,
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "login_user": "Admin",
            "name": "Send alerts to BMC",
            "operations": [
                {
                    "command": null,
                    "command_type": null,
                    "esc_period": null,
                    "esc_step_from": 1,
                    "esc_step_to": 1,
                    "execute_on": null,
                    "host_groups": null,
                    "inventory": null,
                    "media_type": "BMC_Notification",
                    "message": null,
                    "operation_condition": null,
                    "password": null,
                    "port": null,
                    "run_on_groups": null,
                    "run_on_hosts": null,
                    "script_name": null,
                    "send_to_groups": null,
                    "send_to_users": [
                        "zbx_robot"
                    ],
                    "ssh_auth_type": "password",
                    "ssh_privatekey_file": null,
                    "ssh_publickey_file": null,
                    "subject": null,
                    "templates": null,
                    "type": "send_message",
                    "username": null
                }
            ],
            "pause_in_maintenance": true,
            "recovery_default_message": "Problem has been resolved at {EVENT.RECOVERY.TIME} on {EVENT.RECOVERY.DATE}\nProblem name: {EVENT.NAME}\nHost: {HOST.NAME}\nSeverity: {EVENT.SEVERITY}\n\nOriginal problem ID: {EVENT.ID}\n{TRIGGER.URL}\n",
            "recovery_default_subject": "Resolved: {EVENT.NAME}",
            "recovery_operations": [
                {
                    "command": null,
                    "command_type": null,
                    "execute_on": null,
                    "media_type": "BMC_Notification",
                    "message": null,
                    "password": null,
                    "port": null,
                    "run_on_groups": null,
                    "run_on_hosts": null,
                    "script_name": null,
                    "send_to_groups": null,
                    "send_to_users": [
                        "zbx_robot"
                    ],
                    "ssh_auth_type": "password",
                    "ssh_privatekey_file": null,
                    "ssh_publickey_file": null,
                    "subject": null,
                    "type": "send_message",
                    "username": null
                }
            ],
            "server_url": "https://zabbix-myownbox/zabbix/",
            "state": "present",
            "status": "enabled",
            "timeout": 10,
            "validate_certs": false
        }
    },
    "msg": "Action is up to date: Send alerts to BMC"
}

```

### PR DESCRIPTION
**Pull request by D3DeFi**
##### SUMMARY
Exactly the same behavior as fixed in #82 (zbx.mediatype) now uses `name` instead of `description`.

Fixes #75 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/zabbix_action.py

### CODE REVIEW

**Comment by D3DeFi** [comment]
This will have conflicts with #82 due to both modifying CHANGELOG.md. I will rebase one or the other depending on what gets merged first.

We haven't thought about this very much and will deal with such situation more frequently in the future :/

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## devel
 
+### Bug Fixes:
+#### Modules:
+  - zabbix_action - now correctly selects mediatype for the (normal|recovery|update) operations with Zabbix 4.4 and newer.
```
Comment (sky-joker) [inline CHANGELOG.md:7]: How about add at the end the related for PR or issue link?

Comment (D3DeFi) [reply CHANGELOG.md:7]: Yes, I was counting on rebasing this either way due to #92. Btw we are already hitting conflicts in CHANGELOG.md so maybe we should back off on requirement to add CHANGELOG entries and just sum them up before release.

**Comment by D3DeFi** [comment]
Rebased on top of a #92 and should be ready to go

**Comment by sky-joker** [comment]
How about using `LooseVersion` in the following line?

https://github.com/ansible-collections/community.zabbix/blob/e7bc809715c67e4c8f3735811bcf042321996e65/plugins/modules/zabbix_action.py#L869

The following line had been changed, so

https://github.com/ansible-collections/community.zabbix/pull/90/files#diff-c44ea5ba85bfd91841a4ea44a2d54c54L856

**Comment by D3DeFi** [comment]
> 
> 
> How about using `LooseVersion` in the following line?
> 
> https://github.com/ansible-collections/community.zabbix/blob/e7bc809715c67e4c8f3735811bcf042321996e65/plugins/modules/zabbix_action.py#L869
> 
> The following line had been changed, so
> 
> https://github.com/ansible-collections/community.zabbix/pull/90/files#diff-c44ea5ba85bfd91841a4ea44a2d54c54L856

Done. I wanted to do so when rebasing, but then I thought to myself that #97 will probably handle that for us

**Pull request merged**